nrf_rpc/include/nrf_rpc_cbor.h

```diff
@@ -50,33 +52,11 @@ struct _nrf_rpc_cbor_decoder {
  * significant for the API, other fields are internal.
  */
 struct nrf_rpc_cbor_ctx {
-	/** @brief TinyCBOR encoder. */
-	CborEncoder encoder;
-	struct cbor_buf_writer writer;
-	uint8_t *out_packet;
-};
-
-/** @brief Context for encoding commands, sending commands, receiving responses
- * and parsing them.
- *
- * Initialize it with @ref NRF_RPC_CBOR_ALLOC macro. Only `encoder` and `value`
- * fields are significant for the API, other fields are internal.
- */
-struct nrf_rpc_cbor_rsp_ctx {
+	zcbor_state_t zs[2];
 	union {
-		struct {
-			/** @brief TinyCBOR encoder for encoding command. */
-			CborEncoder encoder;
-			struct cbor_buf_writer writer;
-			uint8_t *out_packet;
-		};
-		struct {
-			/** @brief TinyCBOR value for parsing response. */
-			CborValue value;
-			CborParser parser;
-			struct cbor_buf_reader reader;
-			const uint8_t *in_packet;
-		};
+		uint8_t *packet;
+		uint8_t *out_packet;
```

Comment:
It seems like `out_packet` is not used anywhere. Probably we need only two pointers to the packet. I would use:
`uint8_t *out_packet;` and `const unit8_t rsp_packet;`

Reply:
I have left `packet` and `in_packet`.